SEED-platform/seed
# 2538 feat/self registration

#### Any background context you want to provide?
PR to allow self registration of users first logging on to the site.  Note: This does not include assigning an organization, that will be handled by #2539.  When testing, the tester will be able to reach the home page with a red banner saying the user is unaffiliated with an organization.  Due to the reCAPTCHA domain req's - this can only be tested on `localhost:8000` 

#### What's this PR do?
Includes a self registration form with google reCAPTCHA and email verification link.  

#### How should this be manually tested?
Please contact me for the reCAPTCHA secret key to be included in the tester's `local_untracked`.  Reconfigure the development settings by running `DJANGO_SETTINGS_MODULE=config.settings.dev`. After the dev server is running: 

Scenario 1
- Follow the create account workflow all the way through.  (the verification email will be sent to the console)
- Logout and login with the same credentials used to create the account. 

Scenario 2
- Verify error handling on the create an account form is comprehensive. 
 - un-check reCAPTCHA and try to log in.
 - try entering password values that don't meet authentication validation.
 - try to create an account that already exists.

Scenario 3
- After creating an account, try to login with the credentials not via the confirmation email. (should not be possible).


#### What are the relevant tickets?
#2538
